cache_images/fedora-netavark_packaging.sh

```diff
@@ -64,9 +61,17 @@ EXARG="--exclude=netavark --exclude=aardvark-dns"
 msg "Installing general build/test dependencies"
 bigto $SUDO dnf install -y $EXARG "${INSTALL_PACKAGES[@]}"
 
-msg "Installing netavark-specific toolchain dependencies"
-export CARGO_HOME="/var/cache/cargo"  # must match .cirrus.yml in netavark repo
-$SUDO env CARGO_HOME=$CARGO_HOME cargo install mandown cross
-
 # It was observed in F33, dnf install doesn't always get you the latest/greatest
 lilto $SUDO dnf update -y
+
+msg "Initializing upstream rust environment."
+export CARGO_HOME="/var/cache/cargo"  # must match .cirrus.yml in netavark repo
+$SUDO mkdir -p $CARGO_HOME/bin/
+cd $CARGO_HOME/bin/
+$SUDO curl --fail --silent --location -O --url \
+    "https://static.rust-lang.org/rustup/dist/x86_64-unknown-linux-gnu/rustup-init"
+$SUDO chmod +x rustup-init
+# CI Runtime takes care of recovering $CARGO_HOME/env
+$SUDO env CARGO_HOME=$CARGO_HOME ./rustup-init -y --no-modify-path
+# Configure the sysroot for rustc here, so it remains consistent at test runtime
+curl https://sh.rustup.rs -sSf | $SUDO env CARGO_HOME=$CARGO_HOME sh -s -- -y
```

Comment:
Hmmm, looking at `https://sh.rustup.rs` it appears to duplicate running rustup-init.  There's no reason to do this twice.